internal/xds/bootstrap/tlscreds/bundle_ext_test.go

```diff
@@ -224,6 +231,87 @@ func (s) TestCaReloading(t *testing.T) {
 	}
 }
 
+func (s) TestSPIFFEReloading(t *testing.T) {
```

Comment:
This is testing reloading, but can we also test just the basic success & failure cases in separate tests?

Reply:
Still working on this, commit was getting big so I went on and pushed the rest through

Reply:
Complete